src/main/java/com/monumental/models/Model.java

```diff
@@ -4,14 +4,38 @@
 import org.hibernate.annotations.UpdateTimestamp;
 
 import javax.persistence.*;
+import javax.validation.constraints.NotNull;
 import java.util.Date;
 
 @MappedSuperclass
 public abstract class Model {
 
+    /* Bean Validation Groups */
+
+    // Use this group to signify validation that occurs on record creation
+    public interface New {
+
+    }
+
+    // Use this group to signify validation that occurs after record creation
+    public interface Existing {
+
+    }
+
+    // Use this group to signify validation that occurs on record creation or after record creation
+    public interface NewOrExisting extends New, Existing {
+
+    }
+
+    // Use this group to signify validation of the Id field of the record
+    public interface Identity {
```

Comment:
I know I probably wrote this, but I really don't understand what this is for

Reply:
I added the `Identity` one. I was running into the issue that if I called the `validate` method and the `id` field was in the `NewOrExisting` group I would get a `ConstraintViolationException`.

Reply:
Shouldn't `Existing` work? That's how we had it in 343

Reply:
We talked on Slack:

Essentially the Id field should use `Existing` and `NewOrExisting` on most of the other columns. Then when validating (either in a Controller method parameter or manually), use `New` or `Existing`.

The only difference between `Existing` and `New` will be the ID.

Fixing this in the coming commit.